packages/palette/src/elements/DonutChart/DonutChart.tsx

```diff
@@ -0,0 +1,164 @@
+import { Color } from "../../Theme"
+import { Box } from "../Box"
+import { ChartHoverTooltip } from "../DataVis/ChartHoverTooltip"
+import { coerceTooltip } from "../DataVis/ChartTooltip"
+import { ProvideMousePosition } from "../DataVis/MousePositionContext"
+import { ChartProps } from "../DataVis/utils/SharedTypes"
+import { useGetWrapperWidth } from "../DataVis/utils/useGetWrapperWidth"
+import { useHasEnteredViewport } from "../DataVis/utils/useHasEnteredViewPort"
+import { Sans } from "../Typography"
+
+const colors: Color[] = ["black10", "black30", "black60"]
+
+export interface DonutChartProps extends ChartProps {
+  margin?: number
+}
+
+/**
+ * DonutChart is a component that displays data points with donut shaped arcs.
+ * Good for illustrating numerical proportions.
+ */
+export const DonutChart: React.FC<DonutChartProps> = ({
+  points,
+  margin = space(3),
+}: DonutChartProps) => {
+  const [hoverIndex, setHoverIndex] = useState(-1)
+  const [rest, setRest] = useState(false)
+
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const hasEnteredViewport = useHasEnteredViewport(wrapperRef)
+
+  const width = useGetWrapperWidth(wrapperRef)
+
+  const donutWidth = space(4)
+  const donutLabelMargin = space(2) / Math.sqrt(2)
+
+  const centerX = width / 2 + margin
+  const centerY = width / 2 + margin
+
+  const values = points.map(d => d.value).sort()
+  const angelInterpolator = interpolate(0, 2 * Math.PI)
+
+  const arc = d3_arc()
+    .innerRadius(width / 2 - margin - donutWidth)
+    .outerRadius(width / 2 - margin)
+    .padAngle(Math.PI / 360)
+
+  // virtual donut to use its arc centroids to calculate label position
+  const labelArc = d3_arc()
+    .innerRadius(width / 2 - margin + donutLabelMargin)
+    .outerRadius(width / 2 - margin + donutLabelMargin)
+    .padAngle(Math.PI / 360)
+
+  const pie = d3_pie()
+
+  const tooltip = points.map((point, index) => {
+    const hover = hoverIndex === index
+    return hover ? (
+      <ChartHoverTooltip key={`${index}-hover`}>
+        {coerceTooltip(point.tooltip)}
+      </ChartHoverTooltip>
+    ) : null
+  })
+
+  const labels = pie(values).map((slice, index) => {
+    const [x, y] = labelArc.centroid(slice as any)
+    const axisLabelX = points[index].axisLabelX
+    return (
+      axisLabelX && (
+        <DonutLabelContainer
+          key={index}
+          opacity={rest ? 1 : 0}
+          x={x + centerX}
+          y={y + centerY}
+          center={centerX}
+        >
+          <Sans color="black60" size="2">
+            {coerceTooltip(axisLabelX)}
+          </Sans>
+        </DonutLabelContainer>
+      )
+    )
+  })
+
+  const svg = (
+    <svg key={"svg"} width={width + margin} height={width + margin}>
+      <g transform={`translate(${centerX}, ${centerY})`}>
+        <Spring
+          from={{ num: 0 }}
+          to={hasEnteredViewport ? { num: 1 } : { num: 0 }}
+          delay={500}
+          onRest={() => setRest(true)}
+        >
```

Comment:
Can we upgrade to latest react spring? (Sorry for continuing to raise this, but the latest has hook support while maintaining a backwards-compat API.)

Reply:
I'm happy to leave that for another PR, since we'll need to QA all the existing animations. Would it make sense to add a ticket to the platform backlog?

Reply:
I tried to upgrade `spring` in previous PR but it wasn't super straightforward. Apart from API change the logic has also changed a little bit and 👆 doesn't work. Which to @damassi point makes it more important to upgrade sooner than later.

But to unblock our team sprint work I also prefer to create a ticket (for myself in galleries backlog) and do it in a follow up PR if you all agree.

Reply:
@sepans - sounds good. If possible, please follow up on 1807 in a timely manner or file in Platform and we can take care of it.